### ISSUE TITLE
Plans for supporting parallel upload?

### ISSUE THREAD

**Issue by houqp** [tus/tus-py-client#15]
Hi,

I noticed all the official clients are lacking support for parallel upload. Is it because this feature is not production ready or is it because it's just not implemented?

Thanks

**Comment by ifedapoolarewaju** [comment]
@houqp Hi, I believe it's the latter, and also there's been the question of a valid use case for this. But @Acconut recently deliberated on working on an implementation for [tus-js-client](https://github.com/tus/tus-js-client/) for a start pretty soon.

**Comment by Acconut** [comment]
I can only second what @ifedapoolarewaju said. The feature is ready for use but just has not been implemented yet as we had other priorities first. Being said, I will work on an implementation in the new weeks.

**Comment by houqp** [comment]
Thank you both for the info! Looking forward to the implementation. Let me know if there is anything I can help with.

**Comment by fidelix** [comment]
Any progress here?

**Comment by Acconut** [comment]
@fidelix The tus-js-client offers parallel uploads now but to my knowledge this functionality is not available in tus-py-client yet.

**Comment by fidelix** [comment]
Yeah, I'm using it in tus-js but I also need to use this in a python desktop client.
I looked at the code and it seems straightforward to do parallel uploads with threads or async but I don't have the bandwidth at the moment, sorry. Hopefully someone else will be motivated to help.
